services/web/server/src/simcore_service_webserver/studies_dispatcher/_users.py

```diff
@@ -116,11 +116,11 @@ async def _create_temporary_user(request: web.Request):
 
 @log_decorator(_logger, level=logging.DEBUG)
 async def get_or_create_user(
-    request: web.Request, *, is_guest_allowed: bool
+    request: web.Request, *, allow_anonymous_users: bool
 ) -> UserInfo:
     """
     Arguments:
-        is_guest_allowed -- if True, it will create a temporary GUEST account
+        allow_anonymous_users -- if True, it will create a temporary GUEST account
```

Comment:
I find this renaming a little bit confusing to be honnest.
we have anonymous, guest, user and tester.
now anonymous seems to be like guest but not really.

Reply:
@sanderegg if you look at user roles, it includes both: anonymous and guest.

An anonymous user is an user w/o login. Then, under certain conditions, it gets promoted to a GUEST, which is a logged in a has minimal options.

Therefore the same user starts as ANONYMOUS but is promoted to GUEST. That is all :-)

Reply:
@pcrespov I know there are both, that is what I wrote above. But then if I get it correct then the previous implementation was confusing? Then it's ok, if a ANONYMOUS user gets promoted then I guess it is all good. thanks for precision.